knowhere/index/VecIndexFactory.cpp

```diff
@@ -100,23 +100,23 @@ VecIndexFactory::CreateVecIndex(const IndexType& type, const IndexMode mode) {
         }
 #ifdef KNOWHERE_GPU_VERSION
         case IndexMode::MODE_GPU: {
-            auto gpu_device = -1;  // TODO: remove hardcode here, get from invoker
+            auto gpu_device = 0;  // TODO: remove hardcode here, get from invoker
             if (type == IndexEnum::INDEX_FAISS_BIN_IDMAP) {
                 return std::make_shared<knowhere::BinaryIDMAP>();
             } else if (type == IndexEnum::INDEX_FAISS_BIN_IVFFLAT) {
                 return std::make_shared<knowhere::BinaryIVF>();
             } else if (type == IndexEnum::INDEX_FAISS_IDMAP) {
-                return std::make_shared<knowhere::GPUIDMAP>(gpu_device);
+                return std::make_shared<knowhere::IDMAP>();
             } else if (type == IndexEnum::INDEX_FAISS_IVFFLAT) {
-                return std::make_shared<knowhere::GPUIVF_NM>(gpu_device);
+                return std::make_shared<knowhere::GPUIVF>(gpu_device);
```

Comment:
index with different index_mode should match each other, otherwise CopyCpuToGpu() and CopyGpuToCpu() cannot work.
By default we create IVF_NM for CPU flat, if change here to GPUIVF, they mis-match.